libs/metamask/src/lib/+state/metamask.service.ts

```diff
@@ -0,0 +1,78 @@
+import { Injectable } from '@angular/core';
+import { providers, utils } from 'ethers'
+
+export interface ServiceResponse<T> {
+  statut: boolean,
+  message?: string,
+  data?: T
+}
+
+@Injectable({ providedIn: 'root' })
+export class MetamaskService {
+
+  private metamaskDom = (window as any).ethereum;
```

Comment:
you should at least check that  :
- window exists.
- ethereum is part of window

```typescript
signer?: Signer;

get ethereum() {
  if (typeof window !== 'undefined' && 'ethereum' in window) {
    return (window as any).ethereum;
  }
}

hasAccount() {
  return !!this.account;
}

async getAccount() {
  if (this.account) return this.account;
  const ethereum = this.ethereum;
  if (!ethereum) throw new Error('no-web3-provider');
  if (!ethereum.isMetamask) throw new Error('not-metamask');
  const accounts = await ethereum.request({ method: 'eth_requestAccounts' });
  if (!accounts.length) throw new Error('no-accounts-found');
  this.account = accounts[0];
  this.signer = new providers.Web3Provider(ethereum).getSigner();
  return this.account;
}

async sign(message: string) {
  if (!this.signer) {
    await this.getAccount();
  }
  return this.signer.signMessage(message);
}
```